### PR TITLE
Add tools to go.mod

### PR DESCRIPTION
**Pull request by gabegorelick**
This uses the pattern outlined in https://github.com/go-modules-by-example/index/tree/master/010_tools to version tools.

It also fails the build if `go.mod` or `go.sum` are out of date, which ensures that changes to those files are always checked in.

# Issue Type

- Feature Pull Request

## Code contribution checklist

1. [x] I hereby allow the Copyright holder the rights to distribute this piece of
   code under any software license.
1. [ ] The contribution fixes a single existing github issue, and it is linked
   to it.
1. [ ] The code is as simple as possible, readable and follows the idiomatic Go
  [guidelines](https://golang.org/doc/effective_go.html).
1. [ ] All new functionality is covered by automated test cases so the overall
  test coverage doesn't decrease.
1. [ ] No issues are reported when running `make full-test`.
1. [ ] Functionality not applicable to all users should be configurable.
1. [ ] Configurations should be exposed through Lambda function environment
   variables which are also passed as parameters to the
   [CloudFormation](https://github.com/cristim/autospotting/blob/master/cloudformation/stacks/AutoSpotting/template.yaml)
   and
   [Terraform](https://github.com/autospotting/terraform-aws-autospotting/main.tf)
   stacks defined as infrastructure code.
1. [ ] Global configurations set from the infrastructure stack level should also
   support per-group overrides using tags.
1. [ ] Tags names and expected values should be similar to the other existing
   configurations.
1. [ ] Both global and tag-based configuration mechanisms should be tested and
  proven to work using log output from various test runs.
1. [ ] The logs should be kept as clean as possible (use log levels as
  appropriate) and formatted consistently to the existing log output.
1. [ ] The documentation is updated to cover the new behavior, as well as the
   new configuration options for both stack parameters and tag overrides.
1. [ ] A code reviewer reproduced the problem and can confirm the code
  contribution actually resolves it.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -1,15 +1,7 @@
-github.com/aws/aws-lambda-go v1.10.0 h1:uafgdfYGQD0UeT7d2uKdyWW8j/ZYRifRPIdmeqLzLCk=
-github.com/aws/aws-lambda-go v1.10.0/go.mod h1:zUsUQhAUjYzR8AuduJPCfhBuKWUaDbQiPOG+ouzmE1A=
```
Comment (gabegorelick) [inline go.sum]: Deletions are from `go mod tidy`.

**Comment by gabegorelick** [comment]
@cristim This is finally ready for review.

File: Makefile [hunk 2]
```diff
@@ -34,18 +34,16 @@ check_deps:                                                  ## Verify the syste
 	@echo "all dependencies satisifed: $(DEPS)"
 .PHONY: check_deps
 
-build_deps:
-	@command -v goveralls || go get github.com/mattn/goveralls
-	@command -v golint || go get golang.org/x/lint/golint
-	@go tool cover -V || go get golang.org/x/tools/cmd/cover
+build_deps:                                                  ## Install all dependencies specified in tools.go
+	@grep _ tools.go | cut -d '"' -f 2 | xargs go install
```
Comment (gabegorelick) [inline Makefile]: I cannot believe there's no better way to tell go to install all modules specified in go.mod...

File: tools.go [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// +build tools
+
+package tools
+
+import (
+	_ "github.com/mattn/goveralls"
```
[the remark this answers is not in the document]
Comment (gabegorelick) [reply tools.go]: Fixed

File: Makefile (hunk 2 above)
Comment (cristim) [reply Makefile]: Doesn't this require network connectivity on every run?

Comment (gabegorelick) [reply Makefile]: No, it uses the module cache. The first may download modules, but subsequent invocations do not.

**Pull request merged**